### PR TITLE
fix e2e test flake issue - e05

### PR DESCRIPTION
**Pull request by jhellar**
# Description

By using bearer token - https://issues.redhat.com/browse/INTLY-9480.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: test/common/grafana_routes.go [hunk 1]
```diff
@@ -78,8 +73,7 @@ func TestGrafanaExternalRouteDashboardExist(t *testing.T, ctx *TestingContext) {
 	if dashboardResp.StatusCode != http.StatusOK {
 		dumpResp, _ := httputil.DumpResponse(dashboardResp, true)
 		t.Logf("dumpResp: %q", dumpResp)
-		t.Skipf("skipping due to known flaky behaviour https://issues.redhat.com/browse/INTLY-6738, got status : %v", dashboardResp.StatusCode)
-		// t.Fatalf("unexpected status code on success request, got=%+v", dashboardResp)
+		t.Fatalf("unexpected status code on success request, got=%+v", dashboardResp)
```
Comment (david-martin) [inline test/common/grafana_routes.go:76]: Makes sense to remove the skip given the flakey login part has been replaced with using a token.

**Comment by david-martin** [comment]
/approve
Happy for this to land once ci passes

**Comment by david-martin** [comment]
@jhellar I wonder if the context used in e2e tests doesn't have the `get` `namespaces` permission?

```
        --- FAIL: TestIntegreatly/integreatly/E05_-_Verify_Grafana_Route_returns_dashboards (1.44s)
grafana_routes.go:75: dumpResp: "HTTP/1.1 403 Forbidden.....
```

**Comment by jhellar** [comment]
@david-martin @HubertStefanski @obrienrobert I've updated the PR, now the CI passes, can you give it one more look?

File: test/common/grafana_routes.go [hunk 2]
```diff
@@ -86,56 +89,78 @@ func TestGrafanaExternalRouteDashboardExist(t *testing.T, ctx *TestingContext) {
...
-		dumpResp, _ := httputil.DumpResponse(dashboardResp, true)
-		t.Logf("dumpResp: %q", dumpResp)
-		t.Fatalf("unexpected status code on success request, got=%+v", dashboardResp)
-	}
 
-	var dashboards []interface{}
-	if err := json.NewDecoder(dashboardResp.Body).Decode(&dashboards); err != nil {
-		t.Fatal("failed to decode grafana dashboards response", err)
+	// there are more secrets associated with a serviceAccount, one of them should be token that can be used to authenticate
+	// try one by one until the request is successful
```
Comment (david-martin) [inline test/common/grafana_routes.go:108]: It seems a shame the specific secret can't be retrieved by name, name regex or labels.
@jhellar Is there no deterministic way to retrieve the right secret?

Comment (jhellar) [reply test/common/grafana_routes.go]: Thanks for review, this should be fixed now.

**Pull request merged**